src/test/java/seedu/address/storage/JsonSerializableAddressBookTest.java

```diff
@@ -44,4 +47,28 @@ public void toModelType_duplicatePersons_throwsIllegalValueException() throws Ex
                 dataFromFile::toModelType);
     }
 
+    @Test
+    public void toModelType_duplicateTags_throwsIllegalValueException() throws Exception {
+        JsonSerializableAddressBook dataFromFile = JsonUtil.readJsonFile(DUPLICATE_TAG_FILE,
+                JsonSerializableAddressBook.class).get();
+        assertThrows(IllegalValueException.class, JsonSerializableAddressBook.MESSAGE_DUPLICATE_TAG,
+                dataFromFile::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidTagFile_throwsIllegalValueException() throws Exception {
+        JsonSerializableAddressBook dataFromFile = JsonUtil.readJsonFile(INVALID_TAG_FILE,
+                JsonSerializableAddressBook.class).get();
+        assertThrows(IllegalValueException.class, dataFromFile::toModelType);
+    }
+
+    @Test
+    public void toModelType_typicalTagsFile_success() throws Exception {
+        JsonSerializableAddressBook dataFromFile = JsonUtil.readJsonFile(TYPICAL_PERSONS_FILE,
+                JsonSerializableAddressBook.class).get();
+        AddressBook addressBookFromFile = dataFromFile.toModelType();
+        AddressBook typicalPersonsAddressBook = TypicalPersons.getTypicalAddressBook();
+        assertEquals(addressBookFromFile, typicalPersonsAddressBook);
+    }
```

Comment:
This test seems to be testing `TYPICAL_PERSONS_FILE` instead of `TYPICAL_TAGS_FILE` 🤔